Switch from multiple mysqls to one mariadb.

dwt opened the pull request:
This has several advantages:
- mariadb uses drastically less memory by default. On my machine each mysql instance uses at least one gig ram, while mariadb uses less than 40 megabytes.
- only one database again halves the resources used
- only one db is faster to start / stop
- only one db is easier to configure / reason about.

lippserd commented on docker-compose.yml:
```diff
@@ -129,20 +116,22 @@ services:
 
-  icingaweb-mysql:
+  mysql:
+    # see https://hub.docker.com/_/mariadb
```
Please remove that comment.

dwt replied:
- [x] done

lippserd commented on docker-compose.yml:
```diff
@@ -129,20 +116,22 @@ services:
     logging: *default-logging
     volumes:
-      - icingaweb-mysql:/var/lib/mysql
+      # - ./mysql:/var/lib/mysql
```
Please remove that line.

dwt replied:
- [x] done

lippserd commented on docker-compose.yml:
```diff
@@ -129,20 +116,22 @@ services:
-  icingadb-mysql:
+  mysql:
   icingaweb:
   icingaweb-mysql:
```
This volume could be removed as well, right? Also, I like things ordered 😆  So please put `mysql` last here.

dwt replied:
- [x] done

lippserd commented on init-mysql.sh:
```diff
@@ -0,0 +1,16 @@
+EOS
+}
+
+create_database_and_user icingaweb icingaweb ${ICINGAWEB_MYSQL_PASSWORD}
```
Looks like defaults for the env vars are missing, i.e. `icingaweb` and `icingadb`.

dwt replied:
I think these are better set in the docker-compose file, so I set it there.

lippserd commented on docker-compose.yml:
```diff
@@ -129,20 +116,22 @@ services:
-      - icingaweb-mysql:/var/lib/mysql
+      # - ./mysql:/var/lib/mysql
+      - mysql:/var/lib/mysql
+      - ./init-mysql.sh:/docker-entrypoint-initdb.d/init-mysql.sh
```
Please move `init-mysql.sh` to `mysql` (rename it from `icingadb-mysql` and remove `privileges.sql`)  and mount `./env/mysql:/docker-entrypoint-initdb.d`.

dwt replied:
- [x] done

dwt commented:
Will do, I think I can get to that at the weekend or around Tuesday.

lippserd commented:
> Will do, I think I can get to that at the weekend or around Tuesday.

 No rush!

dwt commented:
I've incorporated your feedback and rebased on master to make merging easier. It is a bit hard to test currently, because of [#icinga/docker-icingaweb2/78](https://github.com/Icinga/docker-icingaweb2/issues/78), but here's hope the this gets fixed eventually...

dwt commented:
🎉

lippserd commented:
By the way, thanks for the contribution! 👍
